### DISCUSSION TITLE
Tab proposal

### DISCUSSION THREAD

**Discussion by birtles** [birchill/10ten-ja-reader#675] [Ideas]
This is just to let you know about a change I'm thinking about making to the popup.

We're currently working on making this extension usable on touch screens. One part of that includes making it possible to switch dictionaries without using the keyboard. A tabbed popup seems like the obvious way to do that.

However, I think it might make sense to enable this even in non-touchscreen environments. The reasons are:

* A lot of users have no idea there are multiple dictionaries available.
* Even for users who do, there's no indication about which dictionaries have matches.
* Sometimes the order in which we cycle through dictionaries changes. For example, normally we go words → kanji → names, but if we find no word matches and we _do_ find a names match longer than one character we actually switch to names → kanji. At the moment, that's really opaque: you just have to press <kbd>Shift</kbd> or <kbd>Enter</kbd> and see what happens. The tabs make it much more obvious where you are and what's coming next.

So I think it might still be useful to show tabs even for regular desktop users. On touchscreen devices, however, we could make them bigger to be more easily selectable.

Here's the current prototype in action:

https://user-images.githubusercontent.com/1232595/125049886-6b25d300-e0dc-11eb-9f36-fc1913266baa.mp4

(The code is in the [`combined-query`](https://github.com/birchill/10ten-ja-reader/tree/combined-query) branch.)

A few things I'm still not sure about however are:

* Having the tabs at the top seems a little distracting -- they get in between the word you are looking up and the first (and often best) result. I tried putting them on the bottom but because the height of the pop-up changes it didn't work very well. Maybe on the right-hand-side?
* In order to not make the tabs jump around so much I made the width of the pop-up the same for all dictionaries. As a result the pop-up is probably going to be larger in a number of cases.

**Comment by SaltfishAmi** [comment]
It looks great, and I think it will be useful in combination with sticky popup!   SaltfishAmi   Sent from phone

[I updated your comment to trim the quoted email, I hope you don't mind. @birtles]

**Reply by birtles** [reply]
Thank you!

**Comment by alexulter** [comment]
I think I've just faced one downside for tabs. The thing is, I used to quickly look-up kana spelling for words, and it was pretty convenient in terms of UX to have it on the top. But now top side is occupied by tabs, and I actually suffer a bit each time heaving to find the spelling being downwards. 
It would be pretty neat to have a manageable option to hide tabs (or/and move them to the right side of the popup).

**Reply by birtles** [reply]
Thanks! Yes, I was wondering about this. Incidentally, are you on a touch screen device? (The easiest way to tell if 10ten thinks you're on a touchscreen device is if there is an "X" close button in the top-right of the popup.)

**Reply by alexulter** [reply]
Sorry, I forgot to mention it. I'm on PC/Windows (and there's no "X" button on the popup).

**Reply by birtles** [reply]
Ok, thanks for confirming! I've filed https://github.com/birchill/10ten-ja-reader/issues/688 for addressing this.
